pipenv/utils.py

```diff
@@ -297,23 +307,46 @@ def get_requirement(dep):
         markers = None
     # Strip extras from the requirement so we can make a properly parseable req
     dep, extras = pip.req.req_install._strip_extras(dep)
-    # Only operate on local, existing, non-URI formatted paths
-    if (is_file(dep) and isinstance(dep, six.string_types) and
-            not any(dep.startswith(uri_prefix) for uri_prefix in SCHEME_LIST)):
+    matches_uri = any(dep.startswith(uri_prefix) for uri_prefix in SCHEME_LIST)
+    # Only operate on local, existing, non-URI formatted paths which are installable
+    if is_file(dep) and isinstance(dep, six.string_types) and not matches_uri and is_installable_file(dep):
```

Comment:
So this block is getting a bit out of control. Let's try to simplify a little.

 If something is an "installable file" is it not inherently a "file"? Do we need both checks here?

Are we mutating `dep` to be a non-string above? If it is, are we certain whatever type it's being converted to is still a representation of an installable file?

Reply:
1. I had a long answer typed up here until I reread this more carefully and realized you are right. And in fact I am precisely reimplementing the logic that differentiates the two as well.

2. No we never convert `dep` to be a non-string, that was why I introduced additional variables to split out relevant information as needed